llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3436,6 +3433,16 @@ static void combineMetadata(Instruction *K, const Instruction *J,
     K->setMetadata(LLVMContext::MD_callsite,
                    MDNode::getMergedCallsiteMetadata(KCallSite, JCallSite));
   }
+
+  // Merge prof metadata.
+  // Handle separately to support cases where only one instruction has the
+  // metadata.
+  auto JProf = J->getMetadata(LLVMContext::MD_prof);
+  auto KProf = K->getMetadata(LLVMContext::MD_prof);
+  if (!AAOnly && (JProf || KProf)) {
```

Comment:
The old handling was guarded on DoesKMove - what is the implication of removing that?

Reply:
Removing the condition was intentional. In the case that `DoesKMove` is true, the merging of md_prof can be skipped if `J` does not have prof metadata (a minor optimization). I felt that it was simpler to just perform the merge regardless. Let me know if you feel otherwise.